employee-rostering-backend/src/main/java/org/optaweb/employeerostering/domain/employee/EmployeeAvailability.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaweb.employeerostering.domain.employee;
+
+import java.time.Duration;
+import java.time.OffsetDateTime;
+import java.time.ZoneId;
+
+import javax.persistence.Entity;
+import javax.persistence.FetchType;
+import javax.persistence.ManyToOne;
+import javax.persistence.Table;
+import javax.persistence.UniqueConstraint;
+import javax.validation.constraints.AssertTrue;
+import javax.validation.constraints.NotNull;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import org.optaweb.employeerostering.domain.common.AbstractPersistable;
+
+@Entity
+@Table(uniqueConstraints = @UniqueConstraint(columnNames = {"tenantId", "employee_id", "startDateTime", "endDateTime"}))
+public class EmployeeAvailability extends AbstractPersistable {
+
+    @NotNull
+    @ManyToOne(fetch = FetchType.EAGER)
+    private Employee employee;
+
+    @NotNull
+    private OffsetDateTime startDateTime;
+    @NotNull
+    private OffsetDateTime endDateTime;
+
+    @NotNull
+    private EmployeeAvailabilityState state;
+
+    @SuppressWarnings("unused")
+    public EmployeeAvailability() {
+    }
+
+    public EmployeeAvailability(Integer tenantId, Employee employee, OffsetDateTime startDateTime,
+                                OffsetDateTime endDateTime) {
+        super(tenantId);
+        this.employee = employee;
+        this.startDateTime = startDateTime;
+        this.endDateTime = endDateTime;
+    }
+
+    public EmployeeAvailability(ZoneId zoneId, EmployeeAvailabilityView employeeAvailabilityView, Employee employee) {
+        super(employeeAvailabilityView);
+        this.employee = employee;
+        this.startDateTime = OffsetDateTime.of(employeeAvailabilityView.getStartDateTime(),
+                                             zoneId.getRules().getOffset(employeeAvailabilityView.getStartDateTime()));
+        this.endDateTime = OffsetDateTime.of(employeeAvailabilityView.getEndDateTime(),
+                                             zoneId.getRules().getOffset(employeeAvailabilityView.getEndDateTime()));
+        this.state = employeeAvailabilityView.getState();
+    }
+
+    @AssertTrue
+    @JsonIgnore
+    public boolean isValid() {
```

Comment:
The same architectural issue as with the `Contract` class - violating the SRP.

Reply:
Similar to `Contract`, will add TODO.